internal/datanode/compaction/clustering_compactor.go

```diff
@@ -538,22 +542,26 @@ func (t *clusteringCompactionTask) mappingSegment(
 			}
 			remained++
 
-			// currentSize := t.totalBufferSize.Load()
-			if (remained+1)%20 == 0 {
-				currentBufferSize := t.getWrittenMemoryBufferSize()
-				// trigger spill
-				if clusterBuffer.bufferRowNum.Load() > t.plan.GetMaxSegmentRows() || clusterBuffer.writer.IsFull() {
+			if (remained+1)%100 == 0 {
+				t.rowMemoryEstimationOnce.Do(func() {
+					clusterBuffer.writer.IsFull()
+					t.rowMemoryEstimation = int64(clusterBuffer.writer.WrittenMemorySize()) / clusterBuffer.writer.GetRowNum()
```

Comment:
The method may be inaccurate that estimation the row memory, if it has variable-length field, such as JSON, String, Array, etc.

Reply:
Keep the original writing.